temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInternal.java

```diff
@@ -428,6 +429,11 @@ public static WorkflowInfo getWorkflowInfo() {
     return new WorkflowInfoImpl(getRootWorkflowContext().getContext());
   }
 
+  public static <T> T getMemo(String key, Class<T> valueClass, Type valueType) {
+    Payload memo = getRootWorkflowContext().getContext().getMemo(key);
+    return DataConverter.getDefaultInstance().fromPayload(memo, valueClass, valueType);
```

Comment:
Per our design discussion, customer-defined per client DataConverter should be used (Defined in WorkflowClientOptions), not a global static `DataConverter.getDefaultInstance()`